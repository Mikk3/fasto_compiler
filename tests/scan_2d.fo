fun int write_int(int x) = write(x)
fun  [int]  write_1darr( [int]  x) = map(write_int  , x)
fun [[int]] write_2darr([[int]] x) = map(write_1darr, x)

fun int incr(int a, int b) = a + b
fun [int] incr2d ([int] acc,[int] arr) = scan(incr, 0, arr)

fun int writeInt(int n) = write(n)

fun int main() =
  let a = { {1,2,3}, {4,5,6}, {7,8,9} } in 
  let b = scan(incr2d, {0,0,0}, a) in 
  let out = write_2darr(b) in // 1 2 3 5 7 9 10 13 18
  0

  // 1 (1 + 2) (1 + 2 + 3) 
  // (4 + 1) (4 + 1 + 5)