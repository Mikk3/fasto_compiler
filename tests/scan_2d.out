1 3 6 4 9 15 7 15 24